test/test.js

```diff
@@ -351,3 +359,119 @@ test('skipped assertions count towards the plan', function (t) {
 		t.end();
 	});
 });
+
+test('throws and doesNotThrow work with promises', function (t) {
+	var asyncCalled = false;
+	ava(function (a) {
+		a.plan(2);
+		a.throws(delay(Promise.reject(new Error('foo')), 10), 'foo');
+		a.doesNotThrow(delay(Promise.resolve().then(function () {
+			asyncCalled = true;
+		}), 20));
+	}).run().then(function (a) {
+		t.ifError(a.assertError);
+		t.is(a.planCount, 2);
+		t.is(a.assertCount, 2);
+		t.is(asyncCalled, true);
+		t.end();
+	});
+});
+
+test('t.end is called when a promise passed to t.throws hasn\'t resolved yet', function (t) {
```

Comment:
this makes it sound like AVA will call `t.end` itself. I suggest something like:

`waits for t.throws to resolve after t.end is called`

This comment applies to multiple tests.